### DISCUSSION TITLE
Start new sandbox game

### DISCUSSION THREAD

**Discussion by hytacker** [OpenRCT2/OpenRCT2#15656] [Feature requests]
On the main menu of Orct you can select start a new game. Ater that the game shows you all of the parks you can play but al of these have objectives. I know you can make your own park with your own goal. And have the 'just have fun' option selected that has been added to the game. 

Would it be an idea to have the 'play a continuous game' option at the main menu or in the 'select a new game' menu. So that you don't have to alter an existing objective through the cheat code tab. And have this way of playing added to the base game in a clear and unique way.

**Comment by janisozaur** [comment]
"orct" can mean both "original" and "open", can you please use more verbose name?

**Comment by pizza2004** [comment]
I don't really understand this suggestion.  Can't you ignore the objective and just play the park anyway?  It doesn't close the park or force you to stop playing when you win/lose, does it, or am I wrong?

**Reply by TheClaudeQC** [reply]
when you lose, the park will close and guests are always welcome to quit the park, even if they didn't want to leave, even if you re-opened the park

**Comment by HaasJona** [comment]
I agree with pizza2004, this suggestion doesn't make any sense to me. And yes, you can just continue to play regardless if you win or lose the objective.

**Comment by hytacker** [comment]
In some objective's the game forces you to build new rides at a certain rate. Otherwise your park rating will drop. If it drops below a certain point your park will be closed and the game is lost. This limits you in your way of creating a park. I've encounterd more situations in the game whereby the mechanics of the game hold you back. There's never been an option to play a game without a particular objective untill the 'just have fun' option came in. But this has to be selected specificly through the cheat menu. I'm trying to propose and option in which you can directly start a continouos game from the start. Instead of begining a campaign so to speak.

**Comment by pizza2004** [comment]
But when it closes the park can't you just reopen it and continue on like everything is fine?

**Reply by TheClaudeQC** [reply]
guests will still leave like it was closed if you fail a park

**Comment by hytacker** [comment]
@pizza2004 No it can't.. It's the objetive that doesn't let you drop your park rating below 700. I always put alot of time in scenery. This is made impossible that way.

**Comment by pizza2004** [comment]
What you're asking for is sandbox mode, which I believe has already been a planned feature.  Also, I'm not really sure what you're talking about when you say it requires you to build new rides at specific intervals.  I'm pretty sure there are plenty of various ways to keep the park rating up.  If you want to build a lot of complex scenery without worrying about losing, it's possible to enable building while the game is paused.

**Comment by ghost** [comment]
I believe there is already a "Free Play" park with the have fun objective in the scenario menu. I think it's under the "other parks" sub menu.

**Comment by IntelOrca** [comment]
@osconnoisseur I have no idea what you are talking about. Besides I was thinking of the button for sandbox map to have a map creation wizard, with random map generation integration - like OpenTTD.

**Comment by ghost** [comment]
Ah, never mind. I created a scenario for that purpose a long time ago and thought it was part of OpenRCT2. Disregard what I said.

Is our mapgen currently used at all in the game?

**Comment by Krutonium** [comment]
Only if you use the scenario editor.

**Comment by IntelOrca** [comment]
OpenRCT2 / RCT2 currently only comes with one suitable sandbox park: "Build Your Own Six Flags Park". I pondered whether OpenRCT2 could instead automatically generate a new blank park (possibly a random terrain) with a simple dialog to tweak the settings from the main menu. However when I looked at what other games did, they instead tend to offer a set of sandbox maps that the player can choose with maybe a few basic options to tweak how the gameplay or graphics should look.

This might be a better approch. A new sandbox button could be on the main menu which then takes you to a list similar to scenario select, but instead shows you "workbenches". OpenRCT2 would come with some default ones which give the player a blank map and all the vanilla objects, but the player could also download extra workbenches. The window could include a preview and other options to control whether money and research are enabled. Later down the line it could also let you select between vanilla and open graphics.

![image](https://user-images.githubusercontent.com/1482259/156065033-61f56863-bcff-4dd3-8866-ae29113f5a07.png)

Other game examples:

| | | |
|-|-|-|
| ![image](https://user-images.githubusercontent.com/1482259/156067123-9f809957-22b1-4923-8c6a-d3ae017bb0ad.png) | ![image](https://user-images.githubusercontent.com/1482259/156067144-cc441b70-221f-467a-9fa3-8a2f2fe048a2.png) | ![image](https://user-images.githubusercontent.com/1482259/156067206-c9f8c860-21c3-4dd8-836d-b612083f4a75.png) |

**Reply by karst** [reply]
I like this idea better than what I suggested in this comment https://github.com/OpenRCT2/OpenRCT2/pull/18607#issuecomment-1318745160
Perhaps we could combine it and have people select Workbench or Sandbox tags in the scenario builder, for now have them show up in Other Parks tab, but fully implement it as a separate menu (with some included scenarios, or perhaps a tweaked "Build your own Six Flags Park", to begin with). 
Or would it be the plan if those tags are selected that cheats are on & no money automatically? Then the checkbox in options with cheats would also morph to always have cheats on, while in sandbox mode and workbenches cheats are always enabled (even if unchecked in the options menu).

**Reply by IntelOrca** [reply]
Yeah we would probably still just a category to say whether a park is a workbench or not.

**Reply by karst** [reply]
Concern from @spacek531 in discord was that 5 buttons would look odd. A fix to this could be is adding back tabs on the top but for the main catagories of maps. This means `Campaigns` and `Sandbox` and perhaps one for loose scenarios. Under sandbox it could have tabs similar to the campaign for workbenches, although I think this would become redundant if the store becomes a thing where you could just download full scenery groups.

**Reply by karst** [reply]
Recap of today's discussion in discord:
- Remove cheats from the options menu
- Add option to enable cheats on a scenario, this cannot be undone. The save file will be marked as cheated and this will also be listed on the highscore. Any non-cheated high score will override the existing highscore of the scenario. `Completed with cheats by {Signature}.`
- Ability to select to play with or without money when opening a new sandbox map. All "Have fun" maps will be listed in this catagory.
- Save files will show metadata on the side, similar to how OpenLoco does it: a square screenshot of current active viewport, scenario metadata, and if cheats are enabled or not.
- List three tabs in sandbox menu: Pre-made sandbox maps, Object workbenches, Map generator.
- Sandbox maps can be rewards of campaigns, grey'd out with a text below it `This map can be unlocked when playing {Campaign name}.` or something along those lines. Further discussions on campaigns see #20326

**Comment by 733737** [comment]
I think the best solution is to add a scenario in other parks called "Sandbox" which has RMC Hybrid Coaster (add trains to openrct2 official) and RMC Raptor coasters, and make it 256 by 256 with all rct1 path and land types selected, green grass and 4 entrances, one in the middle of each side of the park

**Comment by karst** [comment]
Perhaps it's a good idea to look at what Age of Empires 2 DE does for their skirmish generator. They have presets of parameters that generate different types of "biomes". 
![image](https://user-images.githubusercontent.com/1516893/212939221-30f5a2f7-b4c7-4999-a558-ab1f2b148e0f.png)

One of the biomes could be similar to what `Build your own Six Flags park` does where it generates a landscape around the outside and keeps a flat area on the inside of the map. Where you could obviously also choose the size (perhaps not in units, but rather in small, medium, big, enormous).

**Reply by IntelOrca** [reply]
This looks like something you would add to the random map generator. Currently you can choose a base terrain, but this is more like choosing your theme which could change or preset many of the generation parameters.

**Reply by karst** [reply]
That's basically what that is ye!

**Reply by Realsteel89** [reply]
thats an nice idea i think do it

**Comment by karst** [comment]
I wanted to bring this topic to attention again.
I have recently had a talk with games lead at Atari in which we have discussed several things.
Among this I got some feedback to make Chris Sawyer happy with the project. Because currently he's not quite happy with it.

The most important take away is that Chris Sawyer does not like having a sandbox mode. I think this is because he thinks it takes away challenge from the game, and well it's a game.
I do understand what he means with it though, why he doesn't like OpenRCT2 is because Cheats are so readily available.

I think the best way forward for the project is to do several things mentioned above.
To sum it up:
- Separate campaign from sandbox entirely (Add a 5th large main menu botton)
- Remove cheats from the options menu and only allow it to be used in sandbox mode
- Allow the conversion of a campaign to sandbox if so desired and make it impossible to get on the scoreboard or progress
- The scenario creator will be merged into sandbox with the option to convert a sandbox world into a scenario.
- Custom campaign objects will allow for adding user made scenario packs to the game
- To add more space, we could add campaign pages similar to Minecraft modded's creative menu
- The sandbox tab will have a tab for user created sandbox maps, workbenches and a map generator with several presets (Similar to AoE2DE's skirmish???).

I believe we could satisfy both Chris Sawyer and our community if we do this.
Chris Sawyer because the challenge will be back in his game.
The community so they can also still use the game as a canvas for their imagination.

**Reply by Derpercoaster** [reply]
Isn't bringing the project to Atari's attention a bad idea? I know some people here had a problem with it last time someone did that.

**Reply by 733737** [reply]
We don't exist in a bubble. Atari and Chris Sawyer have been well aware of the project for many years.
